directory-connector/src/com/untangle/app/directory_connector/ActiveDirectoryManagerImpl.java

```diff
@@ -125,19 +127,47 @@ public List<String> getDomains(){
     }
 
     /**
-     * [getAdapter description]
-     * @param  domain [description]
-     * @return        [description]
+     * Retrieves a mapping of LDAP hosts to domains from the settings of active directory adapters.
+     * @return A map where the keys are LDAP hostnames and the values are corresponding domains.
      */
-    public ActiveDirectoryLdapAdapter getAdapter(String domain){
+    public Map<String, String> getDomainMap() {
+        Map<String, String> domainMap = new HashMap<>();
```

Comment:
`getDomains()` uses a LinkedList, I wonder whether we should be using a LinkedHashMap.